src/main/java/athleticli/ui/Parser.java

```diff
@@ -857,6 +835,36 @@ public static ArrayList<DietGoal> parseDietGoalSetEdit(String commandArgs) throw
         }
     }
 
+    private static ArrayList<DietGoal> initializeIntermmediateDietGoals(String[] commandArgs) throws AthletiException {
+        String[] nutrientAndTargetValue;
+        String nutrient;
+        int targetValue;
+
+        Goal.Timespan timespan = parsePeriod(commandArgs[0]);
+
+        ArrayList<DietGoal> dietGoals = new ArrayList<>();
+        Set<String> recordedNutrients = new HashSet<>();
+
+        for (int i = 1; i < commandArgs.length; i++) {
+            nutrientAndTargetValue = commandArgs[i].split("/");
+            nutrient = nutrientAndTargetValue[0];
+            targetValue = Integer.parseInt(nutrientAndTargetValue[1]);
```

Comment:
If the string isn't a valid integer, this will throw a `NumberFormatException`. Consider wrapping this in a try-catch block and throwing a user-friendly AthletiException if an invalid number is provided.

Reply:
Hi, thanks for the suggestion.
There is an outer function that deals with the number format exception that is calling this function.
This function is created due to refactoring.